raster/r.kappa/testsuite/test_r_kappa.py

```diff
@@ -0,0 +1,296 @@
+        cls.ref_1 = tempname(10)
+        cls.runModule(
+            "r.in.ascii",
+            input=os.path.join(cls.data_dir, "ref_2.ascii"),
+            output=cls.ref_1,
+            quiet=True,
+        )
+        cls.class_1 = tempname(10)
+        cls.runModule(
+            "r.in.ascii",
+            input=os.path.join(cls.data_dir, "class_2.ascii"),
+            output=cls.class_1,
+            quiet=True,
+        )
+        cls.per_class = {
+            "producer": [
+                100.0,
+                100.0,
+                100.0,
+                100.0,
+                100.0,
+                "NA",
+            ],
+            "user": ["NA", "NA", "NA", "NA", "NA", 100.0],
+            "cond.kappa": ["NA", "NA", "NA", "NA", "NA", 0.0],
+        }
+
+    @classmethod
+    def tearDownClass(cls):
+        """Remove temporary data"""
+        cls.del_temp_region()
+        # cls.runModule("g.remove", flags="f", type="raster", name=cls.ref_1)
+        # cls.runModule("g.remove", flags="f", type="raster", name=cls.class_1)
+
+    def match(self, pat, ref):
+        if pat == "NA" or ref == "NA":
+            return pat == ref
+        try:
+            val = float(pat)
+        except ValueError:
+            return False
+        return ref - val < 0.001
+
+    # Characteristics of the matrix
+    # Overall Cohen Kappa value: 0.0
+    # Observation count: 25
+    # Correct count: 0
+    # Overall accuracy: 0.0
+    # CAT TP  TN  FP  FN  User Producer
+    # 0   0   17  0   8   -    0
+    # 1   0   17  0   8   -    0
+    # 2   0   21  0   4   -    0
+    # 3   0   24  0   1   -    0
+    # 4   0   21  0   4   -    0
+    # 9   0   0   25  0   0    -
+
+    def test_standard_output(self):
+        out = read_command(
+            "r.kappa",
+            reference=self.ref_1,
+            classification=self.class_1,
+            flags="w",
+            quiet=True,
+        )
+        rows = out.split("\n")
+
+        # Matrix part
+        self.assertEqual(rows[10], " M     0\t0\t0\t0\t0\t0\t0\t0")
+        self.assertEqual(rows[11], " A     1\t0\t0\t0\t0\t0\t0\t0")
+        self.assertEqual(rows[12], " P     2\t0\t0\t0\t0\t0\t0\t0")
+        self.assertEqual(rows[13], " 2     3\t0\t0\t0\t0\t0\t0\t0")
+        self.assertEqual(rows[14], "       4\t0\t0\t0\t0\t0\t0\t0")
+        self.assertEqual(rows[15], "       9\t4\t8\t8\t4\t1\t0\t25")
+        self.assertEqual(rows[16], "Col Sum\t\t4\t8\t8\t4\t1\t0\t25")
+
+        # Per class coefficients
+        for c in range(len(self.per_class["producer"])):
+            vals = rows[20 + c].split(sep=None)
+            self.assertTrue(self.match(vals[1], self.per_class["user"][c]))
+            self.assertTrue(self.match(vals[2], self.per_class["producer"][c]))
+            self.assertTrue(self.match(vals[3], self.per_class["cond.kappa"][c]))
+
+        # Kappa value
+        vals = rows[28].split(sep=None)
```

Comment:
Given the examples in documentation and tutorials, I think it is generally understood that `split()` operates on whitespace (or at least spaces) and/or that it is the same as as `split(sep=None)`.

The [str.split](https://docs.python.org/3/library/stdtypes.html?highlight=split#str.split) documentation says:

> If sep is not specified or is None, a different splitting algorithm is applied: runs of consecutive whitespace are regarded as a single separator, and the result will contain no empty strings at the start or end if the string has leading or trailing whitespace. Consequently, splitting an empty string or a string consisting of just whitespace with a None separator returns [].

Using `"a\tb".split()` gives `['a', 'b']`.

Reply:
Thanks.